examples/advanced/torque-controls.html

```diff
@@ -0,0 +1,188 @@
+            box-shadow: none;
+            color: #fff;
+            cursor: pointer;
+            display: inline-flex;
+            font: 500 12px/20px 'Roboto';
+            margin: 0;
+            padding: 6px 12px;
+            transition: all 0.2s ease;
+            vertical-align: top;
+        }
+
+        #map {
+            flex: 1;
+        }
+    </style>
+</head>
+
+<body>
+    <div id="sidebar">
+        <h1>Torque</h1>
+        <h2>Controls</h2>
+        <div>
+            <input type="button" id="js-play-btn" value="PLAY">
+            <input type="button" id="js-stop-btn" value="STOP">
+            <input type="button" id="js-pause-btn" value="PAUSE">
+        </div>
+        <h2>Duration</h2>
+        <div>
+            <input type="range" id="js-duration-range" min="1" max="60" step="1">
+            <span id="js-duration-span"></span> 10 seconds
+        </div>
+
+        <h2>Time</h2>
+        <div>
+            <input type="range" id="js-time-range" min="0" max="1" step="0.01">
+        </div>
+    </div>
+
+    <div id="map"></div>
+    <script>
+        const map = new mapboxgl.Map({
+            container: 'map',
+            style: {
+                version: 8,
+                sources: {},
+                layers: [{
+                    id: 'background',
+                    type: 'background',
+                    paint: {
+                        'background-color': 'black'
+                    }
+                }]
+            },
+            center: [0, 0],
+            zoom: 0,
+            dragRotate: false
+        });
+
+        const animStart = '2018-04-11T12:00:00Z';
+        const animEnd = '2018-04-12T12:00:00Z';
+
+        const source = new carto.source.GeoJSON(generateData(), {
+            dateColumns: ['sim_time']
+        });
+
+        function generateData() {
+            const features = [];
+            const length = 200;
+            const n = 50;
+            const x0 = 0;
+            const y0 = 0;
+            let id = 1;
+            const min_st = new Date(animStart);
+            const max_st = new Date(animEnd);
+            for (let step = 1; step <= n; ++step) {
+                features.push({
+                    type: 'Feature',
+                    geometry: {
+                        type: 'Point',
+                        coordinates: [step / n * length + x0 - length / 2, y0]
+                    },
+                    properties: {
+                        cartodb_id: id++,
+                        sim_time: (min_st.getTime() + (max_st - min_st) * step / n)
+                    }
+                });
+            }
+            return {
+                type: 'FeatureCollection',
+                features
+            };
+        }
+
+
+        const viz = new carto.Viz(`
+            @duration: 10
+            @torque: torque($sim_time, @duration, fade(0.2, 0.2))
+            width: linear($sim_time) * 100 + 1
+            strokeWidth: 0
+            filter: @torque`
```

Comment:
any special reason for so many variables?

Reply:
I dont know, just copied an existing map

Reply:
I would remove them, they are unnecessary complexity